evalml/tests/component_tests/test_imputer.py

```diff
@@ -199,32 +199,30 @@ def test_imputer_empty_data(data_type):
     imputer = Imputer()
     imputer.fit(X, y)
     transformed = imputer.transform(X, y)
-    assert_frame_equal(transformed, expected, check_dtype=False)
+    assert_frame_equal(transformed, expected, check_dtype=False, check_index_type=False)
```

Comment:
The index type shouldn't change! When we create the empty dataframes in the test, the index type is `Index`. What happened is that before we would always call `reset_index` so the type would change to `Int64Index`. Since we don't reset the index, the index type should be preserved as `Index`. 

I'm creating `expected` with the correct index type as opposed to setting `check_index_type=False`. I think that's clearer so thanks for the question!